aten/src/ATen/cuda/tunable/GemmCublasLt.h

```diff
@@ -0,0 +1,595 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/cuda/CUDADataType.h>
+#include <ATen/cuda/CUDABlas.h>
+#include <ATen/cuda/tunable/TunableOp.h>
+#include <ATen/cuda/tunable/GemmCommon.h>
+#include <c10/cuda/CUDACachingAllocator.h>
+#include <c10/util/StringUtil.h>
+
+namespace at::cuda::tunable {
+
+template <typename T>
+int GetBatchFromParams(const GemmParams<T>* params) {
```

Comment:
Are these all placeholders or some kind of property that is known statically?

Reply:
They're known statically (e.g batch=1) or returned dynamically (e.g. `return params->batch`) depending on the type of gemm being used here. The TunableOp people did it this way with function overloads for the different type of gemms they support https://github.com/pytorch/pytorch/pull/123987#discussion_r1586865957 and I kept it the same as much as possible